src/generators/typescript/presets/CommonPreset.ts

```diff
@@ -0,0 +1,145 @@
+import { TypeScriptRenderer } from '../TypeScriptRenderer';
+import { TypeScriptPreset } from '../TypeScriptPreset';
+
+import { getUniquePropertyName, FormatHelpers, DefaultPropertyNames } from '../../../helpers';
+import { CommonModel } from '../../../models';
+
+export interface TypeScriptCommonPresetOptions {
+  marshal: boolean;
+  unmarshal: boolean;
+}
+
+/**
+ * Render `marshal` function based on model
+ */
+function renderMarshal({ renderer, model }: {
+  renderer: TypeScriptRenderer,
+  model: CommonModel,
+}): string {
+  const properties = model.properties || {};
+  const propertyKeys = [...Object.entries(properties)];
+  const marshalProperties = propertyKeys.map(([prop, propModel]) => {
+    const camelCasedProp = FormatHelpers.toCamelCase(prop);
```

Comment:
You shouldn't use the `FormatHelpers.toCamelCase` function, because someone can pass another format casing in the `renderer.nameProperty` function - use this last.

Reply:
https://github.com/asyncapi/modelina/blob/master/src/generators/typescript/TypeScriptRenderer.ts#L150

Reply:
Right, good catch! Also, line 23+24 needs to use the property name as is (without formatting) otherwise validation fails. Changed.